File: twentyfive/src/test/java/uk/chaoticgoose/adventofcode/twentyfive/day8/Day8Test.java
package uk.chaoticgoose.adventofcode.twentyfive.day8;

import org.junit.jupiter.api.Test;
import uk.chaoticgoose.adventofcode.twentyfive.utils.TestInputLoader;
import uk.chaoticgoose.adventofcode.twentyfive.utils.TestInputLoader.InputDataModifier;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;
import static uk.chaoticgoose.adventofcode.twentyfive.utils.TestInputLoader.InputDataModifier.LONG;
import static uk.chaoticgoose.adventofcode.twentyfive.utils.TestInputLoader.InputDataModifier.SHORT;

class Day8Test {
    private final TestInputLoader testInputLoader = new TestInputLoader();
    private final InputParser inputParser = new InputParser();
    private final DaySolution underTest = new DaySolution();

    @Test
    void part1Short() {
        assertThat(underTest.part1(inputParser.parse(data(SHORT)), 10)).isEqualTo(40L);
    }

    @Test
    void part1Long() {
        assertThat(underTest.part1(inputParser.parse(data(LONG)), 1000)).isEqualTo(129564L);
    }

    private List<String> data(InputDataModifier inputDataModifier) {
        return testInputLoader.load(8, inputDataModifier);
    }
}

File: src/test/kotlin/twenty_four/day8/Day8Test.kt
package twenty_four.day8

import assertk.assertThat
import assertk.assertions.isEqualTo
import loadData
import org.junit.jupiter.api.Test

class Day8Test {
    private val parser = InputParser()

    @Test
    fun `part 1 long`() {
        assertThat(part1(parser.parse(data()))).isEqualTo(291)
    }

    @Test
    fun `part 1 short`() {
        assertThat(part1(parser.parse(shortData()))).isEqualTo(14)
    }

    private fun data(modifier: String? = null) = loadData(2024, 8, modifier)

    private fun shortData() = data("short")
}
